File: src/main/java/com/wxl/dyttcrawler/pipeline/LogPipeline.java
package com.wxl.dyttcrawler.pipeline;

import com.wxl.dyttcrawler.domain.DyttMovie;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import us.codecraft.webmagic.ResultItems;
import us.codecraft.webmagic.Task;

/**
 * Create by wuxingle on 2020/5/10
 * 记录
 */
@Slf4j
@Component
public class LogPipeline extends DyttPipeline<DyttMovie> {

    public LogPipeline() {
        super(DyttMovie.class);
    }

    @Override
    protected void process(DyttMovie movie, ResultItems resultItems, Task task) {
        log.info("process result:{}, {}", movie.getUrl(), movie.getTitle());
    }
}

File: src/main/kotlin/com/wxl/dyttcrawler/pipeline/LogPipeline.kt
package com.wxl.dyttcrawler.pipeline

import com.wxl.dyttcrawler.domain.DyttMovie
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import us.codecraft.webmagic.ResultItems
import us.codecraft.webmagic.Task

/**
 * Create by wuxingle on 2021/10/11
 * log
 */
@Component
class LogPipeline : DyttPipeline<DyttMovie>(DyttMovie::class.java) {

    companion object {
        private val log = LoggerFactory.getLogger(LogPipeline::class.java)
    }

    override fun process(obj: DyttMovie, resultItems: ResultItems, task: Task) {
        log.info("process result:{}, {}", obj.url, obj.title)
    }

}
